### PR TITLE
Switched docs theme to pytorch-theme

### PR DESCRIPTION
**Pull request by vfdev-5**
Deployement to check the rendering:
- https://vfdev-5.github.io/functorch/

### CODE REVIEW

**Comment by zou3519** [comment]
@vfdev-5 sorry, looking at the rendering reminded me of the actual reason why I didn't use the pytorch theme. The top bar links to various things that are unrelated to functorch!

For examples, tutorials links to PyTorch tutorials, not functorch tutorials...

**Comment by vfdev-5** [comment]
Ok, I can customize that quickly. Do you want to change the topleft logo by something "functorch" ?

**Comment by zou3519** [comment]
Does the theme look OK if we delete the entire top bar? (and the logo?)

Also, there are things missing, like the + in the below image:
image
image

**Comment by vfdev-5** [comment]
I won't remove the header just replace titles and links. Like in pytorch.org/ignite

As for toctree and list expansion I can check that. 
On the other hand I do not insist on switch, if you prefer rtd theme, let's just refresh it

**Comment by zou3519** [comment]
> I won't remove the header just replace titles and links. Like in pytorch.org/ignite

That SGTM, I didn't realize it was easy to replace those.

> As for toctree and list expansion I can check that. On the other hand I do not insist on switch, if you prefer rtd theme, let's just refresh it

I think it'll be good to be consistent with the PyTorch theme, especially because eventually we do want to make sure our docs render nicely under the PyTorch theme. So let's try to make the pytorch theme work for us if it is not too much work.

> Ok, I can customize that quickly. Do you want to change the topleft logo by something "functorch" ?

That would be nice, if we had a graphic designer lol.

File: docs/source/functorch.rst [hunk 1]
```diff
@@ -11,7 +11,9 @@ Function Transforms
 
     grad
     grad_and_value
+    hessian
     jacrev
+    jacfwd
     vmap
     vjp
```
Comment (zou3519) [inline docs/source/functorch.rst:18]: Add jvp too please?

File: docs/source/conf.py [hunk 2]
```diff
@@ -157,44 +154,22 @@
...
-    'css_files': [
-        'https://fonts.googleapis.com/css?family=Lato',
-        '_static/css/pytorch_theme.css'
-    ],
+    "collapse_navigation": False,
+    "display_version": True,
+    "logo_only": True,
+    "pytorch_project": "docs",
+    "navigation_with_keys": True,
+    "analytics_id": "UA-117752657-2",
```
Comment (zou3519) [inline docs/source/conf.py:170]: Is this PyTorch's google analytics ID?

**Review by zou3519 — APPROVED** [review]
One really minor comment. Also, it's really difficult to tell that this is functorch documentation because the logo says "PyTorch", so at some point (doesn't need to be in this PR) we should try to replace it with text that says functorch or something

File: docs/source/conf.py (hunk 2 above)
Comment (vfdev-5) [reply docs/source/conf.py:170]: Yes, it is the same as in vision: 

https://github.com/pytorch/vision/blob/dad6e6a44d63272459b54175c92ed22955eb4207/docs/source/conf.py#L126

and pytorch:

https://github.com/pytorch/pytorch/blob/b9d4c0c7a9689982feb949f9609cec143c4a7268/docs/source/conf.py#L393

**Comment by vfdev-5** [comment]
@zou3519 I updated the docs:
- added vjp entry
- added `functorch` text logo

File: docs/source/conf.py (hunk 2 above)
Comment (zou3519) [reply docs/source/conf.py:170]: Awesome!

**Review by zou3519 — APPROVED** [review]
Thanks! This looks great

**Pull request merged**